### PR TITLE
Add check command

### PR DESCRIPTION
**Pull request by jameslai**
# Summary

The check command returns exit code 0 or 1, depending if the version requirements for the current directory are met. Best used in conjunction with other tooling for automated installation, etc.

I have tested this in conjunction with a direnv containing:

```
if has asdf; then
  asdf check || asdf install
fi
```

...and it works like magic!

Fixes: #734 

## Other Information

This change goes out of its way to take advantage of another asdf function, `current` by parsing its output (rather than re-implementing roughly 80% of that function). Tests provided to ensure nothing breaks if ever the string parsing fails.

### CODE REVIEW

**Comment by Stratus3D** [comment]
Hi @jameslai, thanks for the PR! The code looks good, but I don't think this functionality merits an entirely new command. Our goal has been to provide a simple tool with an interface makes the common tasks easy, and the hard task possible with a little extra code.

There are a couple options here:

* Add an option to the existing `asdf current` command to exit if any tool specified is not installed. e.g. a `-c/--check` flag. This would eliminate the need for this new command as users could just run `asdf current --check || asdf install` instead.
* Do nothing at all. This code is simple, and the same thing can be achieved outside of asdf in only one line of code:
  ```
  (asdf current 2>&1 | grep "Not installed") && asdf install
  ```

  Basically this code takes all the output of `asdf current` (both STDOUT and STDIN `2>&1`) and looks for the string "Not installed" if found the `grep` command succeeds and triggers `asdf install` to be run. If the string is not found the command fails and `asdf install` is not invoked.

I personally don't see the benefit of making any changes to asdf for this, since it can be done in a single line of shell code.

**Comment by jthegedus** [comment]
This is yet another example we could document in a "recipies" section of the docs. Something I want to tackle once we migrate our docs site.

**Comment by Stratus3D** [comment]
@jthegedus I actually have been keeping track of all the recipes that I have shared so far and my plan was to eventually put them into a free ebook. But, that may not happen, and this may be useful to you as well so I'm going to go ahead and share the gist I have - https://gist.github.com/Stratus3D/cf63bcb79695433075d3ce872a584e83. If you want me to convert it to a Git repo so you can send PRs let me know.
